### ISSUE TITLE
.NET 6 XmlSerializer no longer emits line breaks

### ISSUE THREAD

**Issue by RussKie** [dotnet/runtime#64885]
### Description

This appears to be a behavioural regression in .NET 6 in the way XML is serialized. The following code no longer produces the same result in .NET 6 as it was producing in .NET Framework. .NET Core 3.1 and .NET 5.
```cs
            using TextWriter tw = new StreamWriter(path, false);
            XmlSerializer serializer = new(typeof(MyType));
            serializer.Serialize(tw, obj);
```

I was unable to find any breaking docs related to this.

### Reproduction Steps

<details>
<summary>csproj</summary>

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net472;net5.0;net6.0</TargetFrameworks>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>

</Project>
```

</details>

<details>
<summary>Program.cs</summary>

```cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Xml.Serialization;

namespace XmlSerializerNet6
{
    class Program
    {
        static void Main(string[] args)
        {
            List<TranslationItem> items = new();
            items.Add(new(name: "$this", property: "Text", source: "MyName"));

            List<TranslationCategory> categories = new();

            TranslationCategory category = new(name: "MyForm", sourceLanguage: "en", targetLanguage: "");
            category.Body = new() { TranslationItems = items };
            categories.Add(category);

            TranslationFile file = new()
            {
                 GitExVersion = "33.33.33",
                 SourceLanguage = "en",
                 TargetLanguage = "ru",
                 TranslationCategories = categories
            };

            Serialize(file, "test.xlf");
        }

        public static void Serialize(TranslationFile translation, string path)
        {
            using TextWriter tw = new StreamWriter(path, false);
            XmlSerializer serializer = new(typeof(TranslationFile));
            serializer.Serialize(tw, translation);
        }
    }

    /// <summary>Provides a translation for a specific language.</summary>
    [XmlRoot("xliff")]
    public class TranslationFile
    {
        public TranslationFile()
        {
            Version = "1.0";
            TranslationCategories = new List<TranslationCategory>();
        }

        [XmlAttribute("version")]
        public string Version { get; set; }

        [XmlAttribute("GitExVersion")]
        public string? GitExVersion { get; set; }

        [XmlIgnore]
        public string? SourceLanguage { get; set; }

        [XmlIgnore]
        public string? TargetLanguage { get; set; }

        [XmlElement(ElementName = "file")]
        public List<TranslationCategory> TranslationCategories { get; set; }

    }

    [DebuggerDisplay("{" + nameof(Name) + "}")]
    public class TranslationCategory : IComparable<TranslationCategory>
    {
        public TranslationCategory()
        {
        }

        public TranslationCategory(string name, string? sourceLanguage, string? targetLanguage = null)
        {
            Name = name;
            SourceLanguage = sourceLanguage;
            TargetLanguage = targetLanguage;
        }

        [XmlAttribute("datatype")]
        public string DataType { get; set; } = "plaintext";

        [XmlAttribute("original")]
        public string? Name { get; set; }

        [XmlAttribute("source-language")]
        public string? SourceLanguage { get; set; }

        [XmlAttribute("target-language")]
        public string? TargetLanguage { get; set; }

        [XmlElement(ElementName = "body")]
        public TranslationBody Body { get; set; } = new();

        public int CompareTo(TranslationCategory other) => StringComparer.InvariantCulture.Compare(Name, other.Name);
    }

    public class TranslationBody
    {
        public TranslationBody()
        {
            TranslationItems = new List<TranslationItem>();
        }

        [XmlElement(ElementName = "trans-unit")]
        public List<TranslationItem> TranslationItems { get; set; }
    }

    [DebuggerDisplay("{Name}.{Property}={Value}")]
    public class TranslationItem : IComparable<TranslationItem>, ICloneable
    {
        public TranslationItem()
        {
        }

        public TranslationItem(string name, string property, string source)
        {
            Name = name;
            Property = property;
            Source = source;
        }

        public TranslationItem(string? name, string? property, string? source, string? value)
        {
            Name = name;
            Property = property;
            Source = source;
            Value = value;
        }

        [XmlIgnore]
        public string? Name { get; set; }

        [XmlIgnore]
        public string? Property { get; set; }

        [XmlAttribute("id")]
        public string Id
        {
            get => Name + "." + Property;
            set
            {
                var values = value.Split(new char[] { '.' }, 2);
                Name = values[0];
                Property = values.Length > 1 ? values[1] : "";
            }
        }

        [XmlElement("source")]
        public string? Source { get; set; }

        [XmlElement("target")]
        public string? Value { get; set; }

        public int CompareTo(TranslationItem other)
        {
            int val = string.Compare(Name, other.Name, StringComparison.Ordinal);
            if (val == 0)
            {
                val = string.Compare(Property, other.Property, StringComparison.Ordinal);
            }

            return val;
        }

        object ICloneable.Clone()
        {
            return Clone();
        }

        public TranslationItem Clone()
        {
            return new TranslationItem(Name, Property, Source, Value);
        }
    }
}

```

</details>

### Expected behavior

.NET Framework. .NET Core 3.1 and .NET 5 produce the following:
```xml
<?xml version="1.0" encoding="utf-8"?>
<xliff xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.0" GitExVersion="33.33.33">
  <file datatype="plaintext" original="MyForm" source-language="en" target-language="">
    <body>
      <trans-unit id="$this.Text">
        <source>MyName</source>
      </trans-unit>
    </body>
  </file>
</xliff>
```

### Actual behavior

.NET 6 produces the following:
```xml
<?xml version="1.0" encoding="utf-8"?><xliff xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" version="1.0" GitExVersion="33.33.33"><file datatype="plaintext" original="MyForm" source-language="en" target-language=""><body><trans-unit id="$this.Text"><source>MyName</source></trans-unit></body></file></xliff>
```

### Regression?

Yes, in .NET 6.0.

### Known Workarounds

A previously working code has to be rewritten to something like the following:
```cs
            XmlWriterSettings xmlWriterSettings = new()
            {
                Indent = true
            };
            XmlSerializer serializer = new(typeof(TranslationFile));
            using XmlWriter xmlWriter = XmlWriter.Create(path, xmlWriterSettings);
            serializer.Serialize(xmlWriter, translation);
```

However this may not always be possible (e.g., XML serialisation lives in a 3P component) or practical.

**Comment by stephentoub** [comment]
Presumably this is https://github.com/dotnet/runtime/pull/56524 and thus "by design".
@StephenMolloy?

**Comment by RussKie** [comment]
Thank you. If it is an intended change it needs to be formally documented as a [breaking change](https://github.com/dotnet/docs/issues/new/choose).

**Comment by Wraith2** [comment]
I thought that indentation was only for human consumers and that being a serialization format it was nice to have and not essential. What does the lack of indentation break other than readability?

**Comment by kasperk81** [comment]
* suddenly api default has changed to output minified instead of indented content, without updating the breaking change doc.
* the usage is not as simple as changing `false` to `true` but using a different api `XmlWriter.Create` with settings

xml apis have enough usage to avoid breaking changes like toggling defaults for performance reason.. code concerned with performance should already be using Indent=false.

**Comment by Wraith2** [comment]
It's a change but why is is a _breaking_ change? What is it breaking? What exception occurs as a result of there being no whitespace?

**Comment by Wraith2** [comment]
So it broke users looking at the code but not any programs that use the data? is that right?

**Comment by RussKie** [comment]
It is misguided to think that xml is only intended for non-human consumption.
It may also be used for storing user settings, which users can modify (irrespective how usable or not this is in your opinion).
It may also be used as inputs for diff'ing tools, intended for human verification.
Developers find elaborate way to use (and abuse) SDKs. The change of the output format is a breaking change.

In the specific case where this regression was observed, it broke the CI tooling that verified no localisable resources changed. This tooling worked for years, and now it doesn't.

A slight OT. I don't quite agree that all code is/must be concerned with performance. There are plenty of cases where readability and maintainability trump performance, and there are cases where the assembly language is the right tool.
For like 20 years one had to go out of way to remove indents, and now it's flipped.

________________________________

**Comment by opcfoundation-org** [comment]
This change broke my code too, We generate XML files that are stored in git and we rely on git diff to check for unexpected changes. The serialization is done by a NuGet package that I can't easily get a fix for. This is definitely a breaking change that needs more justification than a "performance improvement".

**Comment by mindlink** [comment]
This is a breaking change, for better or for worse.

We rely on an external library that produces XML and then modifies that XML as text on the assumption that line breaks exist.

That assumption is awkward - because it serializes the XML and then manipulates it immediately after and it assumes that the underlying framework that serializes the XML is not going to suddenly change its mind about how it serializes it.

We're not the authors of the library and so changing it is non-trivial. We cannot currently move to net 6 with this change in behaviour.

**Comment by ChristophLindemann** [comment]
Please note, it is not only the indentation that has changed, but worse: it does now add a UTF8 BOM
See #63585

**Comment by StephenMolloy** [comment]
Fixed by #73731
